src/Database/Connection.php

```diff
@@ -142,13 +153,52 @@ public function __construct(array $config)
             'cacheMetaData',
             'cacheKeyPrefix',
         ]));
-        $this->_driver = $this->createDriver($config['driver'] ?? '', $driverConfig);
+
+        $primary = $replica = $this->createDriver($this->_config['drivers'][static::ROLE_PRIMARY]);
+        if (isset($this->_config['drivers'][static::ROLE_REPLICA])) {
+            $replica = $this->createDriver($this->_config['drivers'][static::ROLE_REPLICA]);
```

Comment:
Could there be multiple replicas that an application uses?

Reply:
Where would we randomize it? We want the same driver attached to a Query so changing it with every call to `getDriver()` doesn't seem right.

If we randomly select one when creating a connection then long-running CLI apps will only use 1 replica.